st2api/st2api/controllers/resource.py

```diff
@@ -171,92 +187,87 @@ def _get_all(self, exclude_fields=None, sort=None, offset=0, limit=None, query_o
             else:
                 filters['__'.join(v.split('.'))] = filter_value
 
-        extra = {
-            'filters': filters,
-            'sort': sort,
-            'offset': offset,
-            'limit': limit
-        }
-        LOG.info('GET all %s with filters=%s' % (pecan.request.path, filters), extra=extra)
-
         instances = self.access.query(exclude_fields=exclude_fields, **filters)
         if limit == 1:
             # Perform the filtering on the DB side
             instances = instances.limit(limit)
 
-        if limit:
-            pecan.response.headers['X-Limit'] = str(limit)
-        pecan.response.headers['X-Total-Count'] = str(instances.count())
-
         from_model_kwargs = from_model_kwargs or {}
-        from_model_kwargs.update(self._get_from_model_kwargs_for_request(request=pecan.request))
+        from_model_kwargs.update(self._get_from_model_kwargs_for_request(**kwargs))
 
         result = []
         for instance in instances[offset:eop]:
             item = self.model.from_model(instance, **from_model_kwargs)
             result.append(item)
 
-        return result
+        resp = Response(json=result)
+        resp.headers['X-Total-Count'] = str(instances.count())
+        if limit:
+            resp.headers['X-Limit'] = str(limit)
+
+        return resp
 
-    def _get_one(self, id, exclude_fields=None):
+    def _get_one(self, id, exclude_fields=None, **kwargs):
```

Comment:
While I get the sentiment, I can argue that the damage done to productivity by requiring developer to constantly keep argument list in sync with the spec is much greater. Pretty sure that statical analysis allows you to get the list of keys you're trying to get from the particular kwarg and then compare the list with endpoint parameters.